test/Microsoft.AspNet.Razor.Runtime.Precompilation.Test/PrecompilationTagHelperDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.AspNet.Razor.Test.Internal;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Dnx.Compilation;
+using Microsoft.Dnx.Compilation.CSharp;
+using Microsoft.Dnx.Runtime.Infrastructure;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime.Precompilation
+{
+    public class PrecompilationTagHelperDescriptorFactoryTest : TagHelperDescriptorFactoryTest
+    {
+        private static readonly System.Reflection.TypeInfo CurrentType = 
+            typeof(PrecompilationTagHelperDescriptorFactoryTest).GetTypeInfo();
+        private static readonly Assembly ExecutingAssembly = CurrentType.Assembly;
+        private static readonly string GeneratedAssemblyName = Path.GetRandomFileName() + "." + Path.GetRandomFileName();
+
+        public override ITypeInfo GetTypeInfo(Type tagHelperType)
+        {
+            var paths = new[]
+            {
+                $"{ExecutingAssembly.GetName().Name}.TagHelperDescriptorFactoryTagHelpers.cs",
+                $"{ExecutingAssembly.GetName().Name}.CommonTagHelpers.cs",
+            };
+
+            var compilation = GetCompilation(paths);
```

Comment:
Could cache this in a static to avoid re-parsing the files for every run of a test.